test_helpers/pool_helper.go

```diff
@@ -229,8 +234,11 @@ func SetClusterRO(servers []string, connOpts tarantool.Opts, roles []bool) error
 	return nil
 }
 
-func StartTarantoolInstances(servers []string, workDirs []string,
-	opts StartOpts) ([]TarantoolInstance, error) {
+func StartTarantoolInstances(
+	servers []string,
+	workDirs []string,
+	opts StartOpts,
+	makeDialer func(string) tarantool.Dialer) ([]TarantoolInstance, error) {
```

Comment:
It seems that it more easy just to initialize start options for each instance.
```suggestion
func StartTarantoolInstances(opts []StartOpts) ([]TarantoolInstance, error) {
```